### DISCUSSION TITLE
glow: should I rename my package before submitting to vcpkg?

### DISCUSSION THREAD

**Discussion by rioki** [microsoft/vcpkg#24676] [Q&A] [answered]
With my last port submissions I have seen vcpkg is now checking for popular names and I actually find this super commendable, since there are conflicts bound to happen.

In 2016 I wrote the [OpenGL Object Wrapper: glow](https://github.com/rioki/glow). Now other people quite obviously got the same idea; though no C++ project actually rose to prominence. There is the [GLOW JS library](https://projectglow.io/) and as a result some Linux packages. 

The glow library does not have many users and as a result reaming it would be an option. I would consider it now or never, especially considering that I am dusting off the project, rekindling maintenance (by adding more advanced stuff and Vulkan support) and integrating it into more prominent projects.

Can you please give me some guidance before I get my port rejected and hit a brick wall?

**Comment by ras0219-msft** [comment, marked as answer]
Glad you agree with this important policy and thanks for asking ahead of time!

Yes, we would recommend renaming the package to a longer name (which could still include `glow` -- e.g. `rioki-glow` just like `nlohmann-json`). As you note, while there isn't a current C/C++ conflict, there is the JS library conflict as well as a command line go application https://repology.org/project/glow/versions which has already taken the name `glow` in many Linux distributions. Because vcpkg aims for compatibility across different platforms and ecosystems, we do take significant non-C/C++ projects into account when looking at name conflicts.

**Reply by rioki** [reply]
Thank you, I was already considered the nlohmann-json pattern. That might be a reasonable idea. I have not yet made up my mind if the library really should stand on it's own. But we will see what the next few weeks hold in store.
